### PR TITLE
[Servlet] Implement controller layer

### PR DESCRIPTION
**Pull request by Tim2313**
Here i implemented HashMap into my project.

### CODE REVIEW

File: src/main/java/org/example/constant/UrlPattern.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    public String getUrl() {
+        String finalUrl = "%s%s";
+        return String.format(finalUrl, WAR_NAME, url);
+    }
+
+    // Method for get UrlPattern from the line with Urls
+    public static UrlPattern getBySymbol(String urlPattern) {
+        return URL_PATTERN_STRING_MAP.get(urlPattern);
```
Comment (MLGMAG) [inline src/main/java/org/example/constant/UrlPattern.java]: It is better to remove the substring `/DeveloperApi'` from the input parameter.
Name the input parameter as `url`

File: src/main/java/org/example/constant/UrlPattern.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        URL_PATTERN_STRING_MAP.put(HTML.getUrl(), HTML);
+    }
+
+    UrlPattern(String url) {
+        this.url = url;
+    }
+
+    public String getUrl() {
+        String finalUrl = "%s%s";
+        return String.format(finalUrl, WAR_NAME, url);
```
Comment (MLGMAG) [inline src/main/java/org/example/constant/UrlPattern.java]: It is not ok, to change the initial value of the property that user tries to get

File: src/main/java/org/example/controller/MainController.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    private static final Logger LOGGER = LoggerFactory.getLogger(MainController.class);
+    private final HelloService helloService = HelloService.getInstance();
+    private final NotFoundService notFoundService = NotFoundService.getInstance();
+
+    private static MainController instance;
+
+
+    public Response getHelloType(Arguments arguments) {
+        Response response = new Response();
+        response.setContentType("text/html");
```
Comment (MLGMAG) [inline src/main/java/org/example/controller/MainController.java]: Create separate enum for the context types

File: src/main/java/org/example/controller/MainController.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        String helloResponse = helloService.getHelloResponse();
+        response.setBody(helloResponse);
+        response.setCode(200);
```
Comment (MLGMAG) [inline src/main/java/org/example/controller/MainController.java]: Create separate enum for the response codes

File: src/main/java/org/example/service/PageMapService.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import org.example.controller.MainController;
+import org.example.model.Arguments;
+import org.example.model.Response;
+import org.example.constant.UrlPattern;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+public class PageMapService {
```
Comment (MLGMAG) [inline src/main/java/org/example/service/PageMapService.java]: Move `PageMapService` and `MainServlet` classes into separate package. (`web`)
Rename `PageMapService` to `PathMapper`

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/constant/UrlPattern.java (hunk 1 above)
Comment (Tim2313) [reply src/main/java/org/example/constant/UrlPattern.java]: fixed

File: src/main/java/org/example/constant/UrlPattern.java (hunk 2 above)
Comment (Tim2313) [reply src/main/java/org/example/constant/UrlPattern.java]: fixed

File: src/main/java/org/example/controller/MainController.java (hunk 3 above)
Comment (Tim2313) [reply src/main/java/org/example/controller/MainController.java]: fixed

File: src/main/java/org/example/service/PageMapService.java (hunk 5 above)
Comment (Tim2313) [reply src/main/java/org/example/service/PageMapService.java]: fixed

File: src/main/java/org/example/controller/MainController.java (hunk 4 above)
Comment (Tim2313) [reply src/main/java/org/example/controller/MainController.java]: fixed

File: src/main/java/org/example/constant/UrlPatterns.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package org.example.constant;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public enum UrlPatterns {
```
Comment (MLGMAG) [inline src/main/java/org/example/constant/UrlPatterns.java]: Rename to `UrlPath`

File: src/main/java/org/example/constant/UrlPatterns.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+    private static final Map<String, UrlPatterns> URL_PATTERN_STRING_MAP = new HashMap<>();
+
+    static {
+        URL_PATTERN_STRING_MAP.put(HELLO.getUrl(), HELLO);
+        URL_PATTERN_STRING_MAP.put(JSON.getUrl(), JSON);
+        URL_PATTERN_STRING_MAP.put(HTML.getUrl(), HTML);
+    }
+
+    UrlPatterns(String url) {
```
Comment (MLGMAG) [inline src/main/java/org/example/constant/UrlPatterns.java]: Why the access modifier is absent?

File: src/main/java/org/example/service/RemoveSubstringService.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
+package org.example.service;
+
+public class RemoveSubstringService {
```
Comment (MLGMAG) [inline src/main/java/org/example/service/RemoveSubstringService.java]: Hmm....
Remove substring is a simple statement: `String.replace()`

Why you need a separate class to call this method?
Isn't it easier to simply call this method directly on the String that requires this action?

File: src/main/java/org/example/constant/UrlPatterns.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+        this.url = url;
+    }
+
+    public String getUrl() {
+        return url;
+    }
+
+    // Method for get UrlPattern from the line with Urls
+    public static UrlPatterns getBySymbol(String url) {
```
Comment (MLGMAG) [inline src/main/java/org/example/constant/UrlPatterns.java]: You receive the url with the prefix '/DevelopersApi'.
You should remove this prefix

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/constant/UrlPatterns.java (hunk 6 above)
Comment (Tim2313) [reply src/main/java/org/example/constant/UrlPatterns.java]: fixed

File: src/main/java/org/example/service/RemoveSubstringService.java (hunk 8 above)
Comment (Tim2313) [reply src/main/java/org/example/service/RemoveSubstringService.java]: fixed

File: src/main/java/org/example/constant/UrlPatterns.java (hunk 9 above)
Comment (Tim2313) [reply src/main/java/org/example/constant/UrlPatterns.java]: fixed

File: src/main/java/org/example/constant/UrlPatterns.java (hunk 7 above)
Comment (Tim2313) [reply src/main/java/org/example/constant/UrlPatterns.java]: fixed

**Review by MLGMAG — CHANGES_REQUESTED** [review]
Squash and change commit message to `[Servlet] Implement controller layer`

**Review by MLGMAG — APPROVED** [review]

**Pull request merged**